src/extension.ts

```diff
@@ -129,33 +144,33 @@ export function activate(context: vscode.ExtensionContext) {
         azureIoTExplorer.generateSasTokenForIotHub();
     }));
 
-    context.subscriptions.push(vscode.commands.registerCommand("azure-iot-toolkit.generateSasTokenForDevice", (DeviceItem) => {
-        azureIoTExplorer.generateSasTokenForDevice(DeviceItem);
+    context.subscriptions.push(vscode.commands.registerCommand("azure-iot-toolkit.generateSasTokenForDevice", (deviceNode: DeviceNode) => {
+        azureIoTExplorer.generateSasTokenForDevice(deviceNode.deviceItem);
     }));
 
     context.subscriptions.push(vscode.commands.registerCommand("azure-iot-toolkit.showWelcomePage", () => {
         TelemetryClient.sendEvent(Constants.IoTHubAIShowWelcomePagetEvent, { trigger: "manual" });
         azureIoTExplorer.showWelcomePage();
     }));
 
-    context.subscriptions.push(vscode.commands.registerCommand("azure-iot-toolkit.generateCode", (DeviceItem) => {
-        azureIoTExplorer.generateCode(DeviceItem);
+    context.subscriptions.push(vscode.commands.registerCommand("azure-iot-toolkit.generateCode", (deviceNode: DeviceNode) => {
+        azureIoTExplorer.generateCode(deviceNode.deviceItem);
     }));
 
-    context.subscriptions.push(vscode.commands.registerCommand("azure-iot-toolkit.createModule", (DeviceItem) => {
-        azureIoTExplorer.createModule(DeviceItem);
+    context.subscriptions.push(vscode.commands.registerCommand("azure-iot-toolkit.createModule", (moduleLabelNode: ModuleLabelNode) => {
+        azureIoTExplorer.createModule(moduleLabelNode.deviceNode);
```

Comment:
What if trigger from command palette? moduleLabelNode is undfined

Reply:
fixed